File: customer_app/app/app/src/main/java/com/designated/customer/data/model/CustomerInfo.kt
package com.designated.customer.data.model

import com.google.firebase.Timestamp

/**
 * 고객 정보 데이터 모델
 */
data class CustomerInfo(
    val id: String = "",
    val phoneNumber: String = "",
    val name: String = "",
    val grade: String = "bronze", // bronze/silver/gold/vip
    val points: Int = 0,
    val totalRides: Int = 0,
    val totalSpent: Long = 0L,
    val linkedOfficeId: String = "",
    val primaryOfficeId: String = "", // 최초 연결 사무실 (불변)
    val attributionScore: Int? = null,
    val attributionSource: String? = null, // landing/qr_scan/referral
    val registeredAt: Timestamp = Timestamp.now(),
    val lastRideAt: Timestamp? = null,

    // 사무실 연락처 정보 (신규 추가)
    val officePhone: String = "",      // 사무실 전화번호
    val bankName: String = "",         // 은행명
    val accountNumber: String = "",    // 계좌번호
    val accountHolder: String = "",    // 예금주

    // 주소 정보
    val homeAddress: String = "",      // 집 주소
    val favoriteAddresses: List<String> = emptyList() // 즐겨찾기 주소 목록
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "phoneNumber" to phoneNumber,
            "name" to name,
            "grade" to grade,
            "points" to points,
            "totalRides" to totalRides,
            "totalSpent" to totalSpent,
            "linkedOfficeId" to linkedOfficeId,
            "primaryOfficeId" to primaryOfficeId,
            "attributionScore" to attributionScore,
            "attributionSource" to attributionSource,
            "registeredAt" to registeredAt,
            "lastRideAt" to lastRideAt,
            "officePhone" to officePhone,
            "bankName" to bankName,
            "accountNumber" to accountNumber,
            "accountHolder" to accountHolder,
            "homeAddress" to homeAddress,
            "favoriteAddresses" to favoriteAddresses
        )
    }

    companion object {
        fun fromMap(data: Map<String, Any>): CustomerInfo {
            return CustomerInfo(
                id = data["id"] as? String ?: "",
                phoneNumber = data["phoneNumber"] as? String ?: "",
                name = data["name"] as? String ?: "",
                grade = data["grade"] as? String ?: "bronze",
                points = (data["points"] as? Long)?.toInt() ?: 0,
                totalRides = (data["totalRides"] as? Long)?.toInt() ?: 0,
                totalSpent = (data["totalSpent"] as? Long) ?: 0L,
                linkedOfficeId = data["linkedOfficeId"] as? String ?: "",
                primaryOfficeId = data["primaryOfficeId"] as? String ?: "",
                attributionScore = (data["attributionScore"] as? Long)?.toInt(),
                attributionSource = data["attributionSource"] as? String,
                registeredAt = data["registeredAt"] as? Timestamp ?: Timestamp.now(),
                lastRideAt = data["lastRideAt"] as? Timestamp,
                officePhone = data["officePhone"] as? String ?: "",
                bankName = data["bankName"] as? String ?: "",
                accountNumber = data["accountNumber"] as? String ?: "",
                accountHolder = data["accountHolder"] as? String ?: "",
                homeAddress = data["homeAddress"] as? String ?: "",
                favoriteAddresses = (data["favoriteAddresses"] as? List<*>)?.filterIsInstance<String>() ?: emptyList()
            )
        }
    }
}

File: customer_app_flutter/lib/features/profile/domain/entities/customer_info.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../../../core/domain/converters/timestamp_converter.dart';

part 'customer_info.freezed.dart';
part 'customer_info.g.dart';

/// 손님 프로필 정보 (Kotlin `CustomerInfo.kt:8-85` 포팅).
///
/// **2경로 저장 구조**:
/// 1. `provinces/{p}/cities/{c}/offices/{o}/customers/{uid}` — Anonymous Auth uid 기반 프로필 전체
/// 2. `provinces/{p}/cities/{c}/offices/{o}/customerInfo/{phoneNumber}` — 사무실 연락처 + FCM 토큰 (phoneNumber 기반)
///
/// 모델은 단일. 경로 선택은 Notifier(ProfileNotifier / AuthNotifier)에서 처리.
///
/// **legacy fallback**: 기존 사용자는 `address` 필드에 집 주소를 저장했을 수 있음.
/// [fromFirestore]가 `homeAddress` 우선 + `address` fallback 처리 (Kotlin `fromMap` 일치).
@freezed
class CustomerInfo with _$CustomerInfo {
  const CustomerInfo._();

  const factory CustomerInfo({
    @Default('') String id,
    @Default('') String phoneNumber,
    @Default('') String name,
    @Default('bronze') String grade,
    @Default(0) int points,
    @Default(0) int totalRides,
    @Default(0) int totalSpent,
    @Default('') String linkedOfficeId,
    @Default('') String primaryOfficeId,
    int? attributionScore,
    String? attributionSource,
    @TimestampConverter() DateTime? registeredAt,
    @TimestampConverter() DateTime? lastRideAt,
    @Default('') String officePhone,
    @Default('') String bankName,
    @Default('') String accountNumber,
    @Default('') String accountHolder,
    @Default('') String homeAddress,
  }) = _CustomerInfo;

  factory CustomerInfo.fromJson(Map<String, Object?> json) =>
      _$CustomerInfoFromJson(json);

  /// Firestore 문서 → CustomerInfo (Kotlin `fromMap` 포팅).
  ///
  /// `homeAddress` legacy fallback: 비어 있으면 `address` 필드 참조 (기존 사용자 호환).
  factory CustomerInfo.fromFirestore(
      DocumentSnapshot<Map<String, dynamic>> doc) {
    final data = doc.data() ?? <String, dynamic>{};

    final homeAddr = (data['homeAddress'] as String?)?.isNotEmpty == true
        ? data['homeAddress'] as String
        : (data['address'] as String?) ?? '';

    return CustomerInfo(
      id: data['id'] as String? ?? '',
      phoneNumber: data['phoneNumber'] as String? ?? '',
      name: data['name'] as String? ?? '',
      grade: data['grade'] as String? ?? 'bronze',
      points: (data['points'] as num?)?.toInt() ?? 0,
      totalRides: (data['totalRides'] as num?)?.toInt() ?? 0,
      totalSpent: (data['totalSpent'] as num?)?.toInt() ?? 0,
      linkedOfficeId: data['linkedOfficeId'] as String? ?? '',
      primaryOfficeId: data['primaryOfficeId'] as String? ?? '',
      attributionScore: (data['attributionScore'] as num?)?.toInt(),
      attributionSource: data['attributionSource'] as String?,
      registeredAt: (data['registeredAt'] as Timestamp?)?.toDate(),
      lastRideAt: (data['lastRideAt'] as Timestamp?)?.toDate(),
      officePhone: data['officePhone'] as String? ?? '',
      bankName: data['bankName'] as String? ?? '',
      accountNumber: data['accountNumber'] as String? ?? '',
      accountHolder: data['accountHolder'] as String? ?? '',
      homeAddress: homeAddr,
    );
  }
}
